<!DOCTYPE html>
<html lang="en">
<head>
    <title>ArgumentError</title>
    <meta charset="UTF-8" />
    <link rel="stylesheet" href="../css/reset.css" type="text/css" media="screen" />
<link rel="stylesheet" href="../css/main.css" type="text/css" media="screen" />
<link rel="stylesheet" href="../css/github.css" type="text/css" media="screen" />
<script src="../js/jquery-1.3.2.min.js" type="text/javascript" charset="utf-8"></script>
<script src="../js/main.js" type="text/javascript" charset="utf-8"></script>
<script src="../js/highlight.pack.js" type="text/javascript" charset="utf-8"></script>


    <meta property="og:title" value="ArgumentError">

  
    
    <meta name="description" content="Raised when the arguments are wrong and there isn&#39;t a more specific Exception class.  Ex: passing the wrong number of arguments  [1, 2, 3].">
    <meta property="og:description" content="Raised when the arguments are wrong and there isn&#39;t a more specific Exception class.  Ex: passing the wrong number of arguments  [1, 2, 3].">
  

    <meta name="keywords" content="ArgumentError class">
  
</head>

<body>
    <div class="banner">
        
        <h1>
            <span class="type">Class</span>
            ArgumentError
            
                <span class="parent">&lt;
                    
                    <a href="StandardError.html">StandardError</a>
                    
                </span>
            
        </h1>
        <ul class="files">
            
            <li><a href="../files/ruby/error_c.html">ruby/error.c</a></li>
            
        </ul>
    </div>
    <div id="bodyContent">
        <div id="content">
  
    <div class="description">
      
<p>Raised when the arguments are wrong and there isn&#39;t a more specific <a href="Exception.html"><code>Exception</code></a> class.</p>

<p>Ex: passing the wrong number of arguments</p>

<pre><code>[1, 2, 3].first(4, 5)
</code></pre>

<p><em>raises the exception:</em></p>

<pre><code>ArgumentError: wrong number of arguments (given 2, expected 1)
</code></pre>

<p>Ex: passing an argument that is not acceptable:</p>

<pre><code>[1, 2, 3].first(-4)
</code></pre>

<p><em>raises the exception:</em></p>

<pre><code>ArgumentError: negative array size
</code></pre>

    </div>
  


  


  
  


  


  

  



  

    

    

    


    


    <!-- Methods -->
    
    
    
  
</div>

    </div>
  </body>
</html>
